main_sweeps.py

```diff
@@ -0,0 +1,173 @@
+                    file_path = os.path.join(NUSCENES_ROOT, sensor_data['filename'])
+                    i = 0
+                    if sensor == 'LIDAR_TOP' :
+                        # nuscenes lidar data is stored as (x, y, z, intensity, ring index)
+                        while i < nsweeps_lidar and sensor_token!= '' :
+                            sensor_data = nusc.get('sample_data', sensor_token)
+                            file_path = os.path.join(NUSCENES_ROOT, sensor_data['filename'])
+                            timestamp = round_to_micros(sensor_data['timestamp'])
+                            # Not always exactly 10
+                            if (sensor_data["is_key_frame"] and i!=0) or sample_token== '':
+                                break
+                                #raise AssertionError("Less than 9 sweeps betweeen Lidar samples. Check sweep token no.", sensor_data['token'])
+                            scan = np.fromfile(file_path, dtype=np.float32)
+                            points = scan.reshape((-1, 5))
+
+                            # Transform lidar points from point sensor frame to egovehicle frame
+                            calibration = nusc.get('calibrated_sensor', sensor_data['calibrated_sensor_token'])
+                            egovehicle_R_lidar = quat2rotmat(calibration['rotation'])
+                            egovehicle_t_lidar = np.array(calibration['translation'])
+                            egovehicle_SE3_lidar = SE3(rotation=egovehicle_R_lidar, translation=egovehicle_t_lidar)
+                            points_egovehicle = egovehicle_SE3_lidar.transform_point_cloud(points[:, :3])
+
+                            extract_pc(points_egovehicle, points, output_sensor_path, timestamp)
+
+                            if not os.path.isfile(os.path.join(poses_path, f"city_SE3_egovehicle_{timestamp}.json")) :
+                                ego_pose = nusc.get('ego_pose', sensor_data['ego_pose_token'])
+                                ego_pose_dict = {"rotation": ego_pose["rotation"], "translation": ego_pose["translation"]}
+                                with open(os.path.join(poses_path, f"city_SE3_egovehicle_{timestamp}.json"), 'w') as f:
+                                    json.dump(ego_pose_dict, f)
+
+                            sensor_token = sensor_data['next']
+                    else:
+                        while i < nsweeps_cam and sensor_token != '':
+                            sensor_data = nusc.get('sample_data', sensor_token)
+                            file_path = os.path.join(NUSCENES_ROOT, sensor_data['filename'])
+                            timestamp = round_to_micros(sensor_data['timestamp'])
+                            # Not always exactly 6
+                            if sensor_data["is_key_frame"] and i!=0:
+                                #raise AssertionError("Less than 5 sweeps betweeen camera samples. Check sweep token no.", sensor_data['token'])
```

Comment:
commented out code we can probably remove here

Reply:
Done